ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -228,6 +242,16 @@ public void setOverflow(String overflow) {
     invalidate();
   }
 
+  public void setMaintainVisibleContentPosition(ReactScrollViewMaintainVisibleContentPositionData maintainVisibleContentPositionData) {
+    if (maintainVisibleContentPositionData != null && mMaintainVisibleContentPositionData == null) {
+      getUIManagerModule().addUIManagerEventListener(this);
```

Comment:
Just to clarify, we're using the value of `maintainVisibleContentPositionData` to register/unregister the listener? I'm curious why this is preferred instead of tying it to the class/lifecycle? (i.e class initialisation).

Is this because the invoked callback is expensive?

Reply:
We want to avoid the overhead of tge listener if the prop is not set, iOS had this logic so that is mainly why I added it too.

Reply:
Ah I see, that makes sense 👍